### ISSUE TITLE
Python inconsistent multiprocessing.Process behavior

### ISSUE THREAD

**Issue by IliasAarab** [microsoft/vscode#220267]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 
Version: 1.90.2 (user setup)
Commit: 5437499feb04f7a586f677b155b039bc2b3669eb
Date: 2024-06-18T22:34:26.404Z
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
- OS Version: 
Windows_NT x64 10.0.22621
Steps to Reproduce:

running the below snippet as a Python file in the terminal or within a notebook cell works fine. But running the same snippet within an interactive window fails as the snippet fails to do anything:

```{python}
import multiprocessing
def fn():
    print("here")
process = multiprocessing.Process(target=fn, )
process.start()
process.join()
```

**Comment by amunger** [comment]
I get consistent behavior between vscode notebooks, IW and jupyter notebooks. Are you seeing something different?
![image](https://github.com/user-attachments/assets/68f314c0-dd2c-482f-84c2-5b784ee95295)

**Comment by IliasAarab** [comment]
@amunger sorry for the late response. To make things more concrete, I expanded the code snippet a bit:

```python
# %%
import multiprocessing


def write_digit_to_file(digit, filename, lock):
    with lock:
        with open(filename, "a") as f:
            f.write(f"CPU {multiprocessing.current_process().name} wrote: {digit}\n")


if __name__ == "__main__":
    digits = [1, 2, 3, 4, 5]
    filename = "output.txt"
    lock = multiprocessing.Lock()

    # Clear the file content before starting
    open(filename, "w").close()

    processes = []

    # Create a process for each digit
    for digit in digits:
        p = multiprocessing.Process(
            target=write_digit_to_file, args=(digit, filename, lock)
        )
        processes.append(p)
        p.start()

    # Ensure all processes complete
    for p in processes:
        p.join()

    with open(filename, "r") as f:
        print(f.read())
```

- Running as a python script in vscode:
![image](https://github.com/user-attachments/assets/fe51659f-47f8-4b68-8933-c626b4764b79)
- Running within a vscode interactive window:
![image](https://github.com/user-attachments/assets/f74674f0-4b6a-4ce7-acf5-451497d0bdb6)
- Running within a vscode Jupyter notebook (results are the same with and without putting the code in a __main__ block:
![image](https://github.com/user-attachments/assets/31362579-4bc2-4524-a961-5b1f4fbf1885)
- Running within Google Colab notebook:
![image](https://github.com/user-attachments/assets/f1482a14-fd43-41a7-9e64-a7a8d7e10450)

So now the problem seems to lie within vscode Jupyter notebooks, the file is never populated by the processes. 

I updated my vscode since the last example:

Version: 1.92.1 (user setup)
Commit: eaa41d57266683296de7d118f574d0c2652e1fc4
Date: 2024-08-07T20:16:39.455Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.22621

**Comment by DonJayamanne** [comment]
Please can you try running the same code in Jupyter Notebook
I tried running this against the latest version of Jupyter Lab and Jupyter Notebook and it fails there as well.
I.e. it fails to work in Jupyter Lab, Jupyter Notebook and VS Code
This looks very much like a duplicate of https://github.com/microsoft/vscode-jupyter/issues/12485

> or within a notebook cell works fine. But running the same snippet within an interactive window fails as the snippet fails to do anything:

I think you meant it works in interactive window and fails in a notebook.
